File: csharp/MapleStory.OpenAPI/Src/Dto/Character/CharacterItemEquipmentExceptionalOptionDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐릭터 장비 특별 옵션 정보
    /// </summary>
    public class CharacterItemEquipmentExceptionalOptionDTO
    {
        /// <summary>
        /// STR
        /// </summary>
        [JsonProperty("str")]
        public string Strength { get; set; }

        /// <summary>
        /// DEX
        /// </summary>
        [JsonProperty("dex")]
        public string Dexterity { get; set; }

        /// <summary>
        /// INT
        /// </summary>
        [JsonProperty("int")]
        public string Intelligence { get; set; }

        /// <summary>
        /// LUK
        /// </summary>
        [JsonProperty("luk")]
        public string Luck { get; set; }

        /// <summary>
        /// 최대 HP
        /// </summary>
        [JsonProperty("max_hp")]
        public string MaxHp { get; set; }

        /// <summary>
        /// 최대 MP
        /// </summary>
        [JsonProperty("max_mp")]
        public string MaxMp { get; set; }

        /// <summary>
        /// 공격력
        /// </summary>
        [JsonProperty("attack_power")]
        public string AttackPower { get; set; }

        /// <summary>
        /// 마력
        /// </summary>
        [JsonProperty("magic_power")]
        public string MagicPower { get; set; }

        /// <summary>
        /// 익셉셔널 강화 적용 횟수
        /// </summary>
        [JsonProperty("exceptional_upgrade")]
        public int ExceptionalUpgrade { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterItemEquipmentExceptionalOptionDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * Exceptional equipment option information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterItemEquipmentExceptionalOptionDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterItemEquipmentExceptionalOptionDTO {
    /**
     * STR
     */
    @JsonProperty("str")
    private String strength;

    /**
     * DEX
     */
    @JsonProperty("dex")
    private String dexterity;

    /**
     * INT
     */
    @JsonProperty("int")
    private String intelligence;

    /**
     * LUK
     */
    @JsonProperty("luk")
    private String luck;

    /**
     * Max HP
     */
    @JsonProperty("max_hp")
    private String maxHp;

    /**
     * Max MP
     */
    @JsonProperty("max_mp")
    private String maxMp;

    /**
     * Attack Power
     */
    @JsonProperty("attack_power")
    private String attackPower;

    /**
     * Magic ATT
     */
    @JsonProperty("magic_power")
    private String magicPower;

    /**
     * Number of exceptional boost applications
     */
    @JsonProperty("exceptional_upgrade")
    private int exceptionalUpgrade;
}
